### PR TITLE
Modify S3355(xml): Migrate to LayC

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S3355/xml/rule.adoc [hunk 1]
```diff
@@ -1,42 +1,80 @@
...
 == Resources
 
+=== Documentation
+
+* Struts Docs - https://struts.apache.org/core-developers/web-xml[Web.xml Developpers Guide]
+
+=== Standards
+
 * https://owasp.org/Top10/A05_2021-Security_Misconfiguration/[OWASP Top 10 2021 Category A5] - Security Misconfiguration
 * https://owasp.org/www-project-top-ten/2017/A6_2017-Security_Misconfiguration[OWASP Top 10 2017 Category A6] - Security Misconfiguration
```
Comment (sebastien-andrivet-sonarsource) [inline rules/S3355/xml/rule.adoc:75]: Links are not in the new format.

```suggestion
* OWASP - https://owasp.org/Top10/A05_2021-Security_Misconfiguration/[Top 10 2021 Category A5 - Security Misconfiguration]
* OWASP - https://owasp.org/www-project-top-ten/2017/A6_2017-Security_Misconfiguration[Top 10 2017 Category A6 - Security Misconfiguration]
```

File: rules/S3355/xml/rule.adoc [hunk 2]
```diff
@@ -1,42 +1,80 @@
 == Why is this an issue?
 
-Every filter defined in ``++web.xml++`` file should be used in a ``++<filter-mapping>++`` element. Otherwise such filters are not invoked.
+Struts filters play an important role in web application development. They are
+used to intercept requests and responses from a server and allow developers to
+manipulate them for various purposes such as authentication, logging,
+encryption, and more.
```
Comment (sebastien-andrivet-sonarsource) [inline rules/S3355/xml/rule.adoc:6]: I suggest moving this paragraph before "Why is this an issue?" (summary).

File: rules/S3355/xml/rule.adoc [hunk 3]
```diff
@@ -47,24 +85,6 @@ ifdef::env-github,rspecator-view[]
 
 * "xxx" filter should have a mapping.
 
-
 '''
-== Comments And Links
```
Comment (sebastien-andrivet-sonarsource) [inline rules/S3355/xml/rule.adoc:52]: We do not have a well-defined policy about those comments. Personally, I leave them intact.

File: rules/S3355/xml/rule.adoc [hunk 4]
```diff
@@ -1,42 +1,80 @@
...
-  </filter>
-
-  <filter-mapping>
-     <filter-name>ValidationFilter</filter-name>
-     <url-pattern>/*</url-pattern>
-  </filter-mapping>
+<filter>
+    <filter-name>ValidationFilter</filter-name>
+    <filter-class>com.myco.servlet.ValidationFilter</filter-class>
+</filter> <!-- Noncompliant -->
```
Comment (sebastien-andrivet-sonarsource) [inline rules/S3355/xml/rule.adoc:48]: Maybe, it would be better to put the "Nocompliant" on the `filter-class` line?

**Review by sebastien-andrivet-sonarsource — CHANGES_REQUESTED** [review]
I have some suggestions. There is an error with links.

File: rules/S3355/xml/rule.adoc (hunk 3 above)
Comment (loris-s-sonarsource) [reply rules/S3355/xml/rule.adoc:52]: When they are useless like this conversation, we can remove it, it depends on whether the commends give insights about the rule implementation. Here this data is useless

File: rules/S3355/xml/rule.adoc [hunk 5]
```diff
@@ -1,40 +1,82 @@
+This vulnerability exposes the application to failures of a wide range of
+application-specific features the Strut filter was supposed to perform, such as
+authentication, logging, encryption, and more.
+
```
Comment (loris-s-sonarsource) [inline rules/S3355/xml/rule.adoc:4]: I added this summary, however keep in mind that this section is not supposed to be an introduction explaining the feature: It is supposed to be a summary of the vulnerability.

Here is a compliant example: https://next.sonarqube.com/sonarqube/coding_rules?q=s5131&open=pythonsecurity%3AS5131

File: rules/S3355/xml/rule.adoc [hunk 6]
```diff
@@ -1,40 +1,82 @@
...
 == Why is this an issue?
 
-Every filter defined in ``++web.xml++`` file should be used in a ``++<filter-mapping>++`` element. Otherwise such filters are not invoked.
+Filters are used to intercept requests and responses from a server and allow
+developers to manipulate them.  When a `filter` is declared, but the
```
Comment (loris-s-sonarsource) [inline rules/S3355/xml/rule.adoc:8]:
```suggestion
developers to manipulate them. When a `filter` is declared, but the
```

**Review by sebastien-andrivet-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**